service/src/main/java/bio/terra/externalcreds/controllers/SshKeyApiController.java

```diff
@@ -20,23 +19,24 @@
 @Controller
 public record SshKeyApiController(
     HttpServletRequest request,
-    SamService samService,
+    ExternalCredsConfig externalCredsConfig,
     SshKeyPairService sshKeyPairService,
     AuditLogger auditLogger,
-    ObjectMapper objectMapper)
+    ObjectMapper objectMapper,
+    SamUserFactory samUserFactory)
     implements SshKeyPairApi {
 
   @Override
   public ResponseEntity<Void> deleteSshKeyPair(SshKeyPairType type) {
-    var userId = getUserIdFromSam(request, samService);
+    var samUser = samUserFactory.from(request, externalCredsConfig.getSamBasePath());
```

Comment:
I think it would be better to have a separate component that takes the `ExternalCredsConfig` in its constructor, and provides and API to just pass the request in to get the `samUser`.